src/HtmPixelization.cc

```diff
@@ -0,0 +1,261 @@
+/*
+ * LSST Data Management System
+ * Copyright 2016 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+/// \file
+/// \brief This file contains the HtmPixelization class implementation.
+
+#include "lsst/sphgeom/HtmPixelization.h"
+
+#include "lsst/sphgeom/curve.h"
+#include "lsst/sphgeom/orientation.h"
+
+#include "PixelFinder.h"
+
+
+namespace lsst {
+namespace sphgeom {
+
+namespace {
+
+// Raw HTM root vertex coordinates. UnitVector3d objects are avoided so that
+// library clients cannot run into the static initialization order fiasco.
```

Comment:
Sorry, I still don't understand. I'm imagining the case where you have a client that performs some sphgeom activity as part of the initialization activities for a static object - wouldn't they have to worry about when the constructors for sphgeom statics are called relative to their object, regardless of what the sphgeom constructors actually do?

If the `UnitVector3d` type were trivial (in the formal sense), or had an appropriate `constexpr` constructor, then it feels like the compiler/linker should be able to construct static const objects at compile time and stuff their representations into e.g. the .rodata  section of an executable/library, which would avoid the fiasco.

But I don't know what is guaranteed to happen and what isn't - perhaps my Google-fu is weak - so I stuck to primitive types. Am I missing or misunderstanding something?